flamingpy/utils/viz.py

```diff
@@ -205,53 +197,132 @@ def draw_EGraph(
     fig = plt.figure()
     ax = fig.add_subplot(111, projection="3d")
 
+    # set title
+    title_dict = {
+        "p_phase": "Phase error probabilities",
+        "p_phase_cond": "Conditional phase error probabilities",
+        "hom_val_p": "p-homodyne outcomes",
+        "hom_val_q": "q-homodyne outcomes",
+        "bit_val": "Bit values",
+        "weight": "Weights",
+        "index": "Indices",
+    }
+    name = title_dict.get(label, label)
+    if title and label:
+        ax.set_title(name)
+        ax.title.set_size(plot_params.get("axes.titlesize"))
+
+    # plot graph
+    ax = _plot_EGraph_nodes(ax, egraph, color_nodes, label, name, legend)
+    ax = _plot_EGraph_edges(ax, egraph, color_edges)
+
+    # plot generalities
+    plt.xticks(range(0, 2 * xmax))
+    plt.yticks(range(0, 2 * zmax))
+    ax.set_zticks(range(0, 2 * ymax))
+    ax.set_xlabel("x", labelpad=15)
+    ax.set_ylabel("z", labelpad=15)
+    ax.set_zlabel("y", labelpad=15)
+    if not display_axes:
+        ax.axis("off")
+    plt.tight_layout(pad=5)
+    plt.draw()
+
+    return fig, ax
+
+
+def _plot_EGraph_edges(ax, egraph, color_edges):
+    """Draw the nodes of the graph state represented by the EGraph.
+
+    Args:
+        ax (matplotlib.axes.Axes): the axes to draw the lines in
+        color_edges (bool or string or dict):
+
+            True: color the edges based on the 'color' attribute
+                attached to the node. If unavailable, color nodes grey.
+            string: color all edges with the color specified by the stirng
+            dict: color edges based on attribute and defined colour
+                string by providing a tuple with [attribute, color_dictionary],
+                for example: if the edge attribute "weight" can be +1 or -1,
+                the tuple should be of the form:
+                ``("weight",{-1: minus_color, +1: plus_color})``.
+
+    Returns:
+        A Matplotib Axes object.
+    """
+    # Plotting edges.
+    for edge in egraph.edges:
+        # Color edges based on `color_eges`:
+        #  if `color_edges` is a string use it as color,
+        # using the attribute and color dict if `color_edges` is a tuple(str,dict)
+        # or based on color attribute (when available) if `color_edges` is bool and True;
+        # black otherwise.
+        if isinstance(color_edges, str):
+            color = color_edges
+        elif isinstance(color_edges, tuple):
+            edge_attribute, color_dict = color_edges
+            if not (isinstance(edge_attribute, str) and isinstance(color_dict, dict)):
+                raise ValueError(
+                    "Inappropiate value for `color_edges` argument:"
+                    "Check it complies with the type `tuple(str, dict)`"
```

Comment:
```suggestion
                    "Check that it complies with the type `tuple(str, dict)`,"
```
This may change pending discussion above.